### PR TITLE
Rename TextAnalyticsSubscriptionKeyCredential  => TextAnalyticsApiKeyCredential 

### PR DESCRIPTION
**Pull request by maririos**
And any reference to subscription key => API key

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs [hunk 1]
```diff
@@ -13,18 +13,18 @@ namespace Azure.AI.TextAnalytics.Tests
     public class TextAnalyticsClientLiveTests : RecordedTestBase
     {
         public const string EndpointEnvironmentVariable = "TEXT_ANALYTICS_ENDPOINT";
-        public const string SubscriptionKeyEnvironmentVariable = "TEXT_ANALYTICS_SUBSCRIPTION_KEY";
+        public const string ApiKeyEnvironmentVariable = "TEXT_ANALYTICS_API_KEY";
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: @chidozieononiwu currently the CI uses `TEXT_ANALYTICS_SUBSCRIPTION_KEY` but I am changing it to `TEXT_ANALYTICS_API_KEY`.
What do I have to do to make sure the CI works?

Comment (mssfang) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: You might need to change this here: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/test-resources.json#L84

Comment (mssfang) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: Nvm. Saw your change in the end of PR.

**Review by mssfang — APPROVED** [review]
LGTM!

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: @mssfang !! Yeah, I wasn't sure about it. 
Thanks for checking :)

File: sdk/textanalytics/Azure.AI.TextAnalytics/samples/Sample4_RecognizeEntities.md [hunk 2]
```diff
@@ -3,12 +3,12 @@ This sample demonstrates how to recognize entities in one or more text inputs us
 
 ## Creating a `TextAnalyticsClient`
 
-To create a new `TextAnalyticsClient` to recognize entities in an input text, you need a Text Analytics endpoint and credentials.  You can use the [DefaultAzureCredential][DefaultAzureCredential] to try a number of common authentication methods optimized for both running as a service and development.  In the sample below, however, you'll use a Text Analytics subscription key credential by creating a `TextAnalyticsSubscriptionKeyCredential` object, that if neded, will allow you to update the subscription key without creating a new client.
+To create a new `TextAnalyticsClient` to recognize entities in an input text, you need a Text Analytics endpoint and credentials.  You can use the [DefaultAzureCredential][DefaultAzureCredential] to try a number of common authentication methods optimized for both running as a service and development.  In the sample below, however, you'll use a Text Analytics API key credential by creating a `TextAnalyticsApiKeyCredential` object, that if neded, will allow you to update the API key without creating a new client.
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/samples/Sample4_RecognizeEntities.md]: nit: spelling -> "neded"

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsApiKeyCredential.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            UpdateCredential(apiKey);
+        }
+
+        /// <summary>
+        /// Updates the Cognitive Service API key.
+        /// This is intended to be used when you've regenerated your service API key
+        /// and want to update long lived clients.
+        /// </summary>
+        /// <param name="apiKey">API key to athenticate the service against.</param>
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsApiKeyCredential.cs]: nit: spelling -> athenticate

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/samples/Sample4_RecognizeEntities.md (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/samples/Sample4_RecognizeEntities.md]: Thank u!

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsApiKeyCredential.cs (hunk 3 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsApiKeyCredential.cs]: Thank u!

**Pull request merged**